src/firmware/app/control/python_controller/difference_equation_test.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+"""
+This file contains the unit tests for the DifferenceEquation class
+
+"""
+
+import numpy as np
+import control as ct
+from control.matlab import *
+import unittest
+import differenceequation as de
+import collections
+
+
+class TestDifferenceEquation(unittest.TestCase):
+    def test_simulator_tick(self):
+        """
+        Test that the difference equation simulator follows the correct output sequence
+        as a human-checked sequence with constant input
+        """
+
+        numerator = [1, 1]
+        denominator = [1, 1, 1]
+        interpolation_period = 0.1
+        input_value = 1
+
+        transfer_function = tf(numerator, denominator, interpolation_period)
+        difference_equation = de.DifferenceEquation(transfer_function)
+
+        deSim = de.DifferenceEquationSimulator(difference_equation)
+
+        deSim.tick(input_value)
+        deSim.tick(input_value)
+        deSim.tick(input_value)
+        deSim.tick(input_value)
+        deSim.tick(input_value)
+        deSim.tick(input_value)
+
+        self.assertEqual(deSim.full_output_history[0], 0)
+        self.assertEqual(deSim.full_output_history[1], 1)
+        self.assertEqual(deSim.full_output_history[2], 1)
+        self.assertEqual(deSim.full_output_history[3], 0)
+        self.assertEqual(deSim.full_output_history[4], 1)
+        self.assertEqual(deSim.full_output_history[4], 1)
+
+    def test_simulator_run_for_ticks(self):
+        """
+        This test checks the step response of the DifferenceEquation class vs the discrete transfer function
+        for a specified number of ticks
+        """
+
+        sample_time = 0.01  # [s]
```

Comment:
what's the `# [s]` comment?

Reply:
seconds

Reply:
s means "in seconds"

Reply:
@jonathanlew is correct, I will rename to [seconds] to be more explicit